packages/core/src/modules/oob/repository/OutOfBandRecord.ts

```diff
@@ -44,6 +45,7 @@ export class OutOfBandRecord extends BaseRecord<DefaultOutOfBandRecordTags, Cust
   public autoAcceptConnection?: boolean
   public mediatorId?: string
   public reuseConnectionId?: string
+  public isImplicitInvitation?: boolean
```

Comment:
Curious whether we'd need to store this as a separate property, or whether we may be able to derive it based on the implicit out of band invitation? Should we make the out of band invitation optoinal? Or will that break a lot of existing code?

Reply:
As stated in another comment, at first I wanted to make it optional and use its absence to determine if OOB Record belongs to an implicit invitation.

But besides the breaking code (which is not a lot), I found it useful for each OOB Record to have an unique invitation, especially to store certain parameters like the threadId for the flow (used for instance in `DidExchangeCompleteHandler`). 

We can of course do the same by adding the `threadId` and the `publicDid` as optional properties of OOB Record. This would probably feel more 'natural' and less 'hacky'.